### ISSUE TITLE
Cannot clear email address from provisioning API or occ command

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#37424]
### Steps to reproduce
1. create a user account, e.g. `phil`
2. set an email address, e.g. with the provisioning API:
```
$ curl -X PUT http://phil:phil@172.17.0.1:8080/ocs/v2.php/cloud/users/phil -d key="email" -d value="test@example.org"
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message/>
 </meta>
 <data/>
</ocs>
$ php occ user:list -a email
  - admin: 
  - phil: test@example.org

```
3. try to remove the email address:
```
curl -X PUT http://phil:phil@172.17.0.1:8080/ocs/v2.php/cloud/users/phil -d key="email" -d value=""
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>400</statuscode>
  <message/>
 </meta>
 <data/>
</ocs>
```

Similarly with occ commands:
```
$ php occ user:modify phil email "someone@example.org"
The email address of phil updated to someone@example.org
$ php occ user:list -a email
  - admin: 
  - phil: someone@example.org
$ php occ user:modify phil email ""
The value cannot be empty
```

### Expected behaviour
There should be some way to clear the email address using the Provisioning API  and an occ command.

### Actual behaviour
Errors as above when trying to specify the empty string as the email address.

### Server configuration
Current git core master on Ubuntu 18.04

Note:
1) In the webUI settings, personal, general you can clear and save an empty email address. That calls some other private endpoint.
2) For displayname it works - I can set displayname with e Provisioning API or the occ command, and I can reset it to the empty string.

**Comment by phil-davis** [comment]
Note: old PR #27671 was related to similar stuff and got closed.
Maybe there is a way to allow clearing email addresses through the Provisioning API and occ command - because the webUI certainly does it successfully.

**Comment by phil-davis** [comment]
PR #37425 has a solution for the `occ user:modify` command. That seems to work fine setting email or display name to the empty string.

**Comment by phil-davis** [comment]
Fixed in #37425 (occ user:modify command) and #37427 (Provisioning API)
